client/asset/eth/eth_test.go

```diff
@@ -108,16 +119,34 @@ func (n *testNode) syncProgress(ctx context.Context) (*ethereum.SyncProgress, er
 func (n *testNode) pendingTransactions(ctx context.Context) ([]*types.Transaction, error) {
 	return nil, nil
 }
-func (n *testNode) initiate(opts *bind.TransactOpts, netID int64, initiations []swap.ETHSwapInitiation) (*types.Transaction, error) {
-	return nil, nil
+func (n *testNode) initiate(opts *bind.TransactOpts, netID int64, initiations []dexeth.ETHSwapInitiation) (*types.Transaction, error) {
```

Comment:
Maybe worth noting that this is not concurrent safe because it sets a few values, although the real initiate probably is. It's probably fine that way.

Reply:
Yeah in the real one only the locked funds are updated, and those are protected by a mutex. The tests aren't expected to be concurrent safe are they?

Reply:
That depends if they are used concurrently.  :p

To ensure the product code does not have data races, which have security implications in addition to the potential to cause incorrect behavior and even to panic the app (e.g. concurrent map writes), we run tests with the race detector.  If there is no concurrent use of the method, it doesn't need a sync mechanism, but otherwise it would.

As an aside, having the race detector on with tests helps us be acutely aware of the access patterns of the product code using these stubs.  In many cases we're better of rethinking design than slapping a mutex on everything.  In particular, I'd think these methods don't need synchronization mechanisms (under what circumstances would they be called concurrently?).  A common anti-pattern we face is each test case not cleaning up after itself, and subsequent tests causing data races.  Meddling in internals too much is another one.  It's easy to ignore these issue without the race detector.  I know you weren't suggesting to turn it off though.

I'm always in favor of designing code, both product and test, in such a way that it is lock-free.  That's obv not always possible, just saying.

Reply:
Makes sense. As long as we're not reusing a `testNode` between tests, we'll be good.